test/unit/org/apache/cassandra/metrics/CompactionMetricsTest.java

```diff
@@ -0,0 +1,249 @@
+        this.getCurrentColumnFamilyStore().disableAutoCompaction();
+        for (int i = 0; i < 10; ++i)
+        {
+            this.execute("INSERT INTO %s (pk, ck, a, b) VALUES (" + i + ", 2, 3, 4)");
+        }
+
+        this.getCurrentColumnFamilyStore().forceBlockingFlush();
+        this.getCurrentColumnFamilyStore().forceMajorCompaction();
+
+        assertEquals(1L, compactionManager.getTotalCompactionsCompleted());
+        assertTrue(compactionManager.getTotalBytesCompacted() > 0L);
+        assertTrue(initialCompletedTasks < compactionManager.getCompletedTasks());
+    }
+
+    @Test
+    public void testCompactionMetricsForPendingTasks()
+    {
+        final ColumnFamilyStore cfs = MockSchema.newCFS();
+        final List<SSTableReader> sstables = this.createSSTables(cfs, 10, 0);
+        final Set<SSTableReader> toMarkCompacting = new HashSet<>(sstables.subList(0, 3));
+        final TestCompactionTask tct = new TestCompactionTask(cfs, toMarkCompacting);
+        final long initialPendingTasks = compactionManager.getMetrics().pendingTasks.getValue();
+        try
+        {
+            tct.start();
+            assertTrue(compactionManager.getMetrics().pendingTasksByTableName.getValue().containsKey(cfs.keyspace.getName()));
+            assertTrue(compactionManager.getMetrics().pendingTasks.getValue() > initialPendingTasks);
+        }
+        finally
+        {
+            tct.abort();
+        }
+    }
+
+    @Test
+    public void testCompactionMetricsForCompactionsAborted() throws Throwable
+    {
+        createTable("create table %s (id int, id2 int, t text, primary key (id, id2))");
+        execute("insert into %s (id, id2) values (1, 1)");
+        flush();
+
+        for(File dir : getCurrentColumnFamilyStore().getDirectories().getCFDirectories()){
+            DisallowedDirectories.maybeMarkUnwritable(dir);
+        }
+
+        try
+        {
+            getCurrentColumnFamilyStore().forceMajorCompaction();
+        }
+        catch (Exception e){
+            logger.info("Exception will be thrown when compacting with unwritable directories.");
+        }
+        assertTrue(compactionManager.getMetrics().compactionsAborted.getCount() > 0);
+    }
+
+    @Test
+    public void testCompactionMetricsForDroppedSSTables () throws Throwable
+    {
+        // assertTrue(compactionManager.getMetrics().compactionsReduced.getCount() > 0);
+        // assertTrue(compactionManager.getMetrics().sstablesDropppedFromCompactions.getCount() > 0);
+    }
+
+    private List<SSTableReader> createSSTables(ColumnFamilyStore cfs, int count, int startGeneration)
+    {
+        List<SSTableReader> sstables = new ArrayList<>();
+        for (int i = 0; i < count; i++)
+        {
+            long first = i * 10;
+            long last = (i + 1) * 10 - 1;
+            sstables.add(MockSchema.sstable(startGeneration + i, 0, true, first, last, cfs));
+        }
+        cfs.disableAutoCompaction();
```

Comment:
This is just a style thing, but it feels weird to me that this cfs is passed down just to be modified with the sstables this method is returning.